sihl/src/sihl.mli

```diff
@@ -164,6 +164,261 @@ module Web : sig
+      val edit
+        :  Rock.Request.t
+        -> string
+        -> form
+        -> t
+        -> [> Html_types.html ] Tyxml.Html.elt Lwt.t
+    end
+
+    (** A module of type [CONTROLLER] can be used to create a resource with
+        {!resource_of_controller}. Use a controller instead of a service and a
+        view if you need low level control. *)
+    module type CONTROLLER = sig
+      (** [t] is the type of the resource. *)
+      type t
+
+      (** [index name request] returns a list of all resource instances as a
+          response.
+
+          [name] is the name of the resource in plural, for example [orders] or
+          [users]. *)
+      val index : string -> Rock.Request.t -> Rock.Response.t Lwt.t
+
+      (** [new' ?key name request] returns a form to create instances of the
+          resource as a response.
+
+          [name] is the name of the resource in plural, for example [orders] or
+          [users].
+
+          The form data is stored in the flash storage under the [key]. By
+          default, the value is [_form]. *)
+      val new'
+        :  ?key:string
+        -> string
+        -> Rock.Request.t
+        -> Rock.Response.t Lwt.t
+
+      (** [create name schema request] handles the creation of new resource
+          instances and returns the creation result as a response.
+
+          [name] is the name of the resource in plural, for example [orders] or
+          [users]. *)
+      val create
+        :  string
+        -> ('a, 'b, t) Conformist.t
+        -> Rock.Request.t
+        -> Rock.Response.t Lwt.t
+
+      (** [show name request] returns a single resource instance as a response.
+
+          [name] is the name of the resource in plural, for example [orders] or
+          [users]. *)
+      val show : string -> Rock.Request.t -> Rock.Response.t Lwt.t
+
+      (** [edit ?key name request] returns a form to edit resource instances as
+          a response.
+
+          [name] is the name of the resource in plural, for example [orders] or
+          [users].
+
+          The form data is stored in the flash storage under the [key]. By
+          default, the value is [_form]. *)
+      val edit
+        :  ?key:string
+        -> string
+        -> Rock.Request.t
+        -> Rock.Response.t Lwt.t
+
+      (** [update name schema request] handles the update of a resource instance
+          and returns the update result as a response.
+
+          [name] is the name of the resource in plural, for example [orders] or
+          [users]. *)
+      val update
+        :  string
+        -> ('a, 'b, t) Conformist.t
+        -> Rock.Request.t
+        -> Rock.Response.t Lwt.t
+
+      (** [delete name request] handles the deletion of a resource instance and
+          returns the deletion result as a response.
+
+          [name] is the name of the resource in plural, for example [orders] or
+          [users]. *)
+      val delete' : string -> Rock.Request.t -> Rock.Response.t Lwt.t
```

Comment:
Can we find a name that does not need a tick?

Reply:
The names of those actions seem to be quite popular. I would like to have new in the name so it doesn't break expectations of someone using rails. I can't think of a better name at them moment, do you have one?

Reply:
This seems to be the best name. Only other I can think of is `remove` which doesn't convey the meaning as well IMO. Let's leave it like this then :)